### PR TITLE
[RFC] vim-patch:7.4.1056

### PR DESCRIPTION
**Pull request by shotat**
Problem:    Don't know why finding spell suggestions is slow.
Solution:   Add some code to gather profiling information.

https://github.com/vim/vim/commit/ca1fe985175385c609f8e06672a1014729aba05c

### CODE REVIEW

**Comment by KillTheMule** [comment]
So wait, https://github.com/neovim/neovim/pull/4870/commits/b9a5f2fc6957f537c3c3ede1c96a3a9dd599a976#diff-a02ad7e0f5d95056cf4bb7f258e1a5afR9325 means that this code won't be compiled anyways unless you change the 0, right? You should at least set it to 1 so the build systems can test this, but somehow I think this should be integrated into the  build system. But if it's still not enabled in some builds, this will be dead code...

Not sure, someone else should chime in on this.

**Comment by shotat** [comment]
OMG

**Comment by jamessan** [comment]
Well, that's the intention of the patch.  It's debug code to help diagnose the performance of the spell checking.  It should only be enabled when it's needed.  There are a few defines like that (`SPELL_PRINTTREE`, `DEBUG_TRIEWALK`, `SPELL_COMPRESS_ALLWAYS`) in the spell checking code.

**Comment by KillTheMule** [comment]
> Well, that's the intention of the patch

I realized that, but what made me think about it is that we won't notice if that code breaks, because it's not active. Point in case, this PR passed all checks, without telling us much.


> There are a few defines like that

Well, if there's precedence it should be fine. Maybe though we should have a "spellcheck" build target that enables all these so we can at least check locally from time to time that this stuff is still alright?

**Comment by jamessan** [comment]
Yeah, changing the `#if 0` to `#ifdef SUGGEST_PROFILE` is probably saner than the current code.

**Comment by jamessan** [comment]
I notice you're merging neovim/master into your PR.  Typically, we use a rebase workflow instead.

**Comment by shotat** [comment]
Oh, sorry, I will fix soon.

**Comment by shotat** [comment]
Now rebase master. Thanks for your help. 🙏

File: src/nvim/spell.c [hunk 1]
```diff
@@ -9319,7 +9319,56 @@ static void suggest_try_special(suginfo_T *su)
   }
 }
 
+// Measure how much time is spent in each state.
+// Output is dumped in "suggestprof".
+
+#ifdef SUGGEST_PROFILE
```
Comment (shotat) [inline src/nvim/spell.c]: Changed to `#ifdef SUGGEST_PROFILE` from `#if 0`.

**Pull request merged**